targets/TARGET_STM/TARGET_STM32F4/TARGET_STM32F429xI/device/TOOLCHAIN_ARM_MICRO/stm32f429xx.sct

```diff
@@ -29,38 +29,49 @@
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 
 #if !defined(MBED_APP_START)
-  #define MBED_APP_START 0x08000000
+  #define MBED_APP_START            0x08000000
 #endif
 
+; 2 MB FLASH (0x200000)
 #if !defined(MBED_APP_SIZE)
-  #define MBED_APP_SIZE 0x200000
+  #define MBED_APP_SIZE             0x200000
 #endif
 
-#define MBED_RAM_START              0x20000000
-#define MBED_RAM_SIZE               0x20000
-#define MBED_VECTTABLE_RAM_START    (MBED_RAM_START)
-#define MBED_VECTTABLE_RAM_SIZE     0x1B0
-#define MBED_CRASH_REPORT_RAM_START (MBED_VECTTABLE_RAM_START + MBED_VECTTABLE_RAM_SIZE)
-#define MBED_CRASH_REPORT_RAM_SIZE  0x100
-#define MBED_RAM0_START             (MBED_CRASH_REPORT_RAM_START + MBED_CRASH_REPORT_RAM_SIZE)
-#define MBED_RAM0_SIZE              (MBED_RAM_SIZE - MBED_VECTTABLE_RAM_SIZE - MBED_CRASH_REPORT_RAM_SIZE)
+;256 KB SRAM (0x40000)
+#if !defined(MBED_RAM_START)
+  #define MBED_RAM_START            0x20000000
+#endif
+
+#if !defined(MBED_RAM_SIZE)
+  #define MBED_RAM_SIZE             0x20000 ; (?)
```

Comment:
I think you have added the "?" because size should be 0x30000....
I don't know where this value come from...
But I have a general question: is this "#if !defined" really needed ?
I understand that, with mbed compilation scripts, we will never use the "default" value defined here ?

Reply:
@jeromecoutant  - Yes I was not sure of size.

Mbed OS tools do provide RAM/ROM sizes, but those are fetched from CMSIS package and not available for all targets. To remove `ifdef` we need to be sure that sizes are available at compilation time, hence I will leave that to example owners.

Please see the template we recommend for linker files - https://github.com/ARMmbed/mbed-os-5-docs/pull/951.
If we follow the template and provide a mechanism to add vector count as well we can considerably reduce the number of linker files and can have common for each category.

Reply:
@jeromecoutant  - Please confirm if the size should be 0x20000 or 0x30000. I also see additional 0x10000 IARM2 in ARM_STD file which is not part of uARM, should be considered when merging both.

Reply:
Confirmed by mail  - 0x30000  is right value 👍